src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -70,23 +70,29 @@ public MonikerProvider(Docset docset)
                     return (error, configMonikers);
                 }
 
-                var fileMonikers = _rangeParser.Parse(fileMonikerRange);
-                var intersection = configMonikers.Intersect(fileMonikers, StringComparer.OrdinalIgnoreCase).ToList();
+                var fileMonikers = _rangeParser.ParseWithInfo(fileMonikerRange);
+                var intersection = configMonikers.Intersect(fileMonikers).ToList();
 
                 // With non-empty config monikers,
                 // warn if no intersection of config monikers and file monikers
                 if (intersection.Count == 0)
                 {
-                    error = Errors.EmptyMonikers($"No moniker intersection between docfx.yml/docfx.json and file metadata. Config moniker range '{configMonikerRange}' is '{string.Join(',', configMonikers)}', while file moniker range '{fileMonikerRange}' is '{string.Join(',', fileMonikers)}'");
+                    error = Errors.EmptyMonikers($"No moniker intersection between docfx.yml/docfx.json and file metadata. Config moniker range '{configMonikerRange}' is '{string.Join(',', configMonikers.Select(x => x.MonikerName))}', while file moniker range '{fileMonikerRange}' is '{string.Join(',', fileMonikers.Select(x => x.MonikerName))}'");
                 }
-                intersection.Sort(Comparer);
+                intersection.OrderBy(x => x.MonikerName, Comparer);
                 return (error, intersection);
             }
 
-            configMonikers.Sort(Comparer);
+            configMonikers.OrderBy(x => x.MonikerName, Comparer);
```

Comment:
this will not change the order of items in configMonikers, the return value is  sorted

Reply:
You are right. why are the tests green here...

Reply:
remove this line